### PR TITLE
feat: Add support for simple button loading via useAsyncHandler

### PR DESCRIPTION
**Pull request by benjackwhite**
## Problem

I find myself often having a button which does something like "Create a new thing" or "Do this quick async thing". Typically we build this into a logic but I've encountered enough cases where I _don't_ want a logic as it is entirely a "do API request and redirect".

## Changes

* Adds a little hook for taking care of this typical loading state use case
* Keeps the LemonButton as it was but details how this can be used in the Storybook
* Pulled out of this PR where it is used for creating and duplicating playlists https://github.com/PostHog/posthog/pull/13056

### CODE REVIEW

**Review by mariusandra — COMMENTED** [review]
No strong opposition here to the idea... though I'd clean up some variable names in this implementation.

However, why you were fearing strong opposition from me is because I am still worried that things _will_ get misused, and given enough time with not enough 👀 on the right things at the right moments, we'll end up building a fully parallel data layer next to the existing one... just because buttons made it so easy. It'll start small, but a few things will lead to another, and suddenly the mess we've gotten into is easy to solve if we "just add xstate" or some other 3rd solution. Give it enough time and we're an XKCD comic 😅.

I see how currently this makes everything nicer though, so feel free to proceed.

However, would [something like this](https://github.com/keajs/kea/pull/152/files#diff-5622465081c38369e7fb16c6b5beb4df862f67fc3216ee39cff81bf9a62097cfR26) help out in any way to bridge the gaps?

File: frontend/src/lib/components/LemonButton/LemonButton.tsx [hunk 1]
```diff
@@ -77,22 +79,42 @@ function LemonButtonInternal(
...
         disableClientSideRouting,
+        onClick,
         ...buttonProps
     }: LemonButtonProps,
     ref: React.Ref<HTMLElement>
 ): JSX.Element {
-    if (loading) {
-        icon = <Spinner monocolor />
-    }
+    const [clickLoading, setClickLoading] = useState<boolean | null>(null)
```
[the remark this answers is not in the document]
Comment (Twixes) [reply frontend/src/lib/components/LemonButton/LemonButton.tsx:87]: `onFooBar` values usually are event handlers 🤔

**Review by Twixes — CHANGES_REQUESTED** [review]
I'm ideologically on board with this (though `asyncActions` looks super interesting), but this feature would be much nicer as a hook
```TypeScript
function useAsyncHandler<E extends React.UIEvent>(
    explicitLoading: boolean | undefined, onEvent: ((e: E) => any | Promise<any>) | undefined
): { loading: boolean, onEvent: (e: E) => void }
```

**Comment by benjackwhite** [comment]
> I'm ideologically on board with this (though `asyncActions` looks super interesting), but this feature would be much nicer as a hook
> 
> ```ts
> function useAsyncHandler<E extends React.UIEvent>(
>     explicitLoading: boolean | undefined, onEvent: ((e: E) => any | Promise<any>) | undefined
> ): { loading: boolean, onEvent: (e: E) => void }
> ```

This is the right answer 👏  . It's ever so slightly more code from the user perspective but is worth it as we don't push bad habits as much and keep the LemonButton nice and clean.

Will update the PR to reflect that.

**Review by Twixes — APPROVED** [review]
👌

**Pull request merged**